### PR TITLE
Character roles

### PR DESCRIPTION
**Pull request by WillDenton1989**
https://trello.com/c/VrqoPvME/78-add-role-to-characters

### CODE REVIEW

File: app/models/character_template.rb [hunk 1]
```diff
@@ -13,6 +13,14 @@ class CharacterTemplate < ApplicationRecord
     HERO = "hero".freeze
   ].freeze
 
+  CHARACTER_ROLES = [
+    MONARCH = "monarch".freeze,
```
Comment (mpanetta) [inline app/models/character_template.rb:17]: These aren't the names that were assigned.

File: app/views/provinces/_province_details.html.erb [hunk 2]
```diff
@@ -48,7 +48,7 @@
   <div class="row mt-2 mx-1 pe-1 align-items-top text-nowrap text-truncate">
     <div class="col-4 text-start">
       <div class="governor-portrait">
-        <%= image_tag("head_large.png", :class => "img-fluid m-0 p-0") %>
+        <%= image_tag("characters/character_monarch.png", :class => "img-fluid m-0 p-0") %>
```
Comment (mpanetta) [inline app/views/provinces/_province_details.html.erb:51]: This isn't how this should work. Each character should have an image based on just them, and whoever happens to be the chancellor, etc. will be selected based on their role for presentation on the appropriate context. E.G. if one of my characters is named bill denton, and has the key "BILL_DENTON_CHARACTER_TEMPLATE" , then the we should have an image in the assets/characters directory called "bill_denton_character_portrait.png", or something substantially similar.

File: spec/factories/character_templates.rb [hunk 3]
```diff
@@ -10,6 +10,7 @@
     gender { CharacterTemplate::GENDERS.sample }
     starting_age { rand(15..80) }
     character_class { CharacterTemplate::CHARACTER_CLASSES.sample }
+    character_role { CharacterTemplate::CHARACTER_ROLES.sample }
```
Comment (mpanetta) [inline spec/factories/character_templates.rb:13]: :+1:

File: spec/models/character_template_spec.rb [hunk 4]
```diff
@@ -20,4 +20,14 @@
       end
     end
   end
+
+  describe "#character_role" do
+    context "when the character_role is not a valid value" do
+      it "raises an exception" do
+        expect do
+          create(:character_template, :character_role => "invalid")
```
Comment (mpanetta) [inline spec/models/character_template_spec.rb:28]: Very nice!

**Comment by mpanetta** [comment]
This looks good. I'd like to see the roles named as asked. I think we can tackle the images in a follow up work as the comment suggests. Otherwise, :shipit:

File: app/models/character_template.rb (hunk 1 above)
Comment (WillDenton1989) [reply app/models/character_template.rb:17]: As per our brief conversation i was under the impression that the specific names we used were not particularly relevant and i changed them to be more concise. The conversation about titles with Gary, which you saw on the whiteboard, was adjacent but not relevant to this work. I understand they are to be used on the back end and have no relevance to the front end, which i qualified in our conversation.

**Pull request merged**